sdk/core/azure-core-amqp/inc/azure/core/amqp/common/completion_operation.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// SPDX - License - Identifier : MIT
+
+#pragma once
+
+#include <condition_variable>
+#include <list>
+#include <mutex>
+#include <tuple>
+
+namespace Azure { namespace Core { namespace _internal { namespace Amqp { namespace Common {
+
+  /** Abstracts an operation sent to azure-c-shared-utility/azure-uamqp.
+   */
+  template <typename CompleteFn, typename WrapperClass> struct CompletionOperation
+  {
+    CompleteFn m_onOperationComplete;
+
+    CompletionOperation(CompleteFn onOperationComplete) : m_onOperationComplete{onOperationComplete}
+    {
+    }
+
+    template <typename... Args2> static void OnOperationFn(void* context, Args2... args)
```

Comment:
I can't use `Args` because the compiler complains. So I had to come up with something else to express the concept of "the arguments to this function".

I'm open to an alternate  variable name.

Reply:
`Arguments`, `T`...
`Args2` rather starts unnecessary questions, just like the one I had.